modules/templates/templates/templates.test.tsx

```diff
@@ -0,0 +1,732 @@
+      act(() => templateEdit[2].click());
+
+      await waitFor(() => {
+        expect(
+          screen.getByTestId("template-detail.field-type-select"),
+        ).toBeInTheDocument();
+      });
+
+      const typeSelectElement = screen.getByTestId(
+        "template-detail.field-type-select",
+      );
+      expect(typeSelectElement).toHaveValue("multiselect");
+
+      await waitFor(() => {
+        expect(
+          screen.queryByTestId("template-detail.edit-field-config"),
+        ).toBeInTheDocument();
+      });
+
+      const configButton = screen.getByTestId(
+        "template-detail.edit-field-config",
+      );
+      act(() => configButton.click());
+
+      const modal = await screen.findByTestId("template-detail.field-modal");
+      expect(modal).toBeInTheDocument();
+
+      const { getByText, queryByText } = within(modal);
+      expect(queryByText("test_type")).not.toBeInTheDocument();
+
+      const optionInputElement = screen.getByTestId(
+        "field-modal.multi-select-input",
+      );
+      await userEvent.type(optionInputElement, "test_type");
+
+      const addOption = screen.getByTestId(
+        "field-modal.multi-select-add-option",
+      );
+      expect(addOption).toBeInTheDocument();
+
+      act(() => addOption.click());
+
+      await waitFor(() => {
+        expect(getByText("test_type")).toBeInTheDocument();
+        expect(getByText("test_type")).toHaveClass("optionTag");
+      });
+    });
+
+    it("Checks modal pops up when delete icon clicked", async () => {
+      render(<TemplateDetail />);
+
+      // Wait for the table to be fully rendered (at least one row is filled)
+      await waitFor(() => {
+        expect(screen.getByTestId("table-row-0"));
+      });
+
+      const templateEdit = screen.getAllByTestId("template-detail.delete");
+      act(() => templateEdit[0].click());
+
+      await waitFor(() => {
+        expect(
+          screen.getByTestId("template-detail.delete-confirmation"),
+        ).toBeInTheDocument();
+      });
+    });
+
+    it("Checks template delete function", async () => {
+      render(<TemplateDetail />);
+
+      await waitFor(() => {
+        expect(screen.getByTestId("table-row-0"));
+      });
+
+      const templateEdit = screen.getAllByTestId("template-detail.delete");
+      act(() => templateEdit[0].click());
+
+      await waitFor(() => {
+        expect(
+          screen.getByTestId("template-detail.delete-confirmation"),
+        ).toBeInTheDocument();
+      });
+
+      const confirmButton = screen.getByTestId("modal.confirm-button");
+      expect(confirmButton).toBeInTheDocument();
+      act(() => confirmButton.click());
+
+      await waitFor(() => {
+        expect(
+          screen.queryByTestId("template-detail.delete-confirmation"),
+        ).not.toBeInTheDocument();
+      });
+
+      // TODO: Make this check dynamic
+      expect(screen.queryByText("Age")).not.toBeInTheDocument();
+    });
+  });
```

Comment:
I miss the pagination tests 

Is easy to test pagination if we mock the max items per page to 1. Or we just mock the API result to return total 5 items, 1, page size 1